@types/jsb.d.ts

```diff
@@ -705,137 +749,797 @@ declare namespace jsb {
          * @en Response code for the in-app messaging API call.
          * @zh 应用内消息传递 API 调用的响应代码。
          */
-        readonly responseCode: number;
+        getResponseCode(): number;
         /**
          * @en Token that identifies the purchase to be acknowledged, if any.
          * @zh 返回标识需要确认的购买的令牌。
          */
-        readonly purchaseToken: string;
+        getPurchaseToken(): string;
+    }
+
+    export interface BillingResultBuilder {
+        setDebugMessage(productType: string): BillingResultBuilder;
+        setResponseCode(productType: number): BillingResultBuilder;
+        build(): BillingResult;
+    }
+    export class BillingResult {
+        static Builder: {
+            new(): BillingResultBuilder
+        };
+        getResponseCode(): number;
+        getDebugMessage(): string;
+        toStr(): string;
+        public static newBuilder(): BillingResultBuilder;
+    }
+
+    export namespace BillingResult {
+        type Builder = BillingResultBuilder;
+    }
+
+    export interface UserChoiceBillingListener {
+        userSelectedAlternativeBilling(userChoiceDetails: UserChoiceDetails): void;
+    }
+    export interface PurchasesUpdatedListener {
+        onPurchasesUpdated(billingResult: BillingResult, purchases: Purchase[]): void;
+    }
+    export interface BillingClientStateListener {
+        onBillingServiceDisconnected(): void;
+        onBillingSetupFinished(
+            billingResult: BillingResult,
+        ): void;
+    }
+
+    export interface PendingPurchasesParamsBuilder {
+        enableOneTimeProducts: () => PendingPurchasesParamsBuilder;
+        enablePrepaidPlans: () => PendingPurchasesParamsBuilder;
+        build: () => PendingPurchasesParams;
+    }
+
+    export class PendingPurchasesParams {
+        private constructor();
+        public static newBuilder(): PendingPurchasesParamsBuilder;
+    }
+
+    export interface ProductBuilder {
+        setProductId: (productID: string) => ProductBuilder;
+        setProductType: (productType: string) => ProductBuilder;
+        build: () => Product;
+    }
+    export class Product {
+        private constructor();
+        public static newBuilder(): ProductBuilder;
```

Comment:
Missing `setProductId` and `setProductType` ?

![Screenshot 2024-12-10 at 11 34 48](https://github.com/user-attachments/assets/6e4fcfad-0ad9-423c-b100-f986b979fca9)